### PR TITLE
Suggested RFC changes

### PR DESCRIPTION
**Pull request by dsnet**
I recommend reviewing each commit individually. The "large" commits that touch many lines are actually fairly mechanical changes that do not change the semantic meaning of the text at all. I included rationale for every change in the description of each commit.

Keeping track of what has been merged (or abandoned):
* ~~(6ad0428) Minor formatting changes~~
* ~~(317ea91) Fix ambiguity about whether final bit padding must be zero~~
* (1a5c031) Make pseudo-code more code-like
* ~~(466f356) Clarify simple and complex prefix codes~~
* ~~(1a4f884) Define the maximum number of bytes transforms may add to a word~~
* (805ff4f) Minor formatting changes
* ~~(04ba9ac) Minor formatting of Section 9.2. and Section 9.3.~~
* ~~(fcd05db) Clarify Section 4.~~
* ~~(2459bb7) Use consistent bit convention in Section 5.~~
* ~~(f9f173c) Clarify section 7.1.~~
* ~~(4d375a2) Clarify Section 7.3~~
* ~~(238d459) Clarify Section 8.~~

### CODE REVIEW

**Comment by szabadka** [comment]
I started reviewing the changes, but I am not finished yet.

I propose that we use this huge pull request for discussion and that you create separate pull requests for each smaller change after the review for them is complete.

As a start, the commit for Section 7.1 (f9f173c...) looks good as is and could be merged.

**Comment by dsnet** [comment]
SGTM. I made a seperate PR for Section 7.1. (f9f173c...)

File: docs/draft-alakuijala-brotli-07.nroff [hunk 1]
```diff
@@ -1719,13 +1774,18 @@ The decoding algorithm that produces the uncompressed data is as follows:
...
             compute context ID, CIDD from CLEN
-            read distance code using HTREED[CMAPD[4 * BTYPE_D + CIDD]]
+            read distance code using HTREED[CMAPD[4*BTYPE_D + CIDD]]
             compute distance by distance short code substitution
-         move backwards distance bytes in the uncompressed data and
-            copy CLEN bytes from this position to the uncompressed
-            stream, or look up the static dictionary word, transform
-            the word as directed, and copy the result to the
-            uncompressed stream
+            if distance code is not zero, and distance is not a
```
Comment (szabadka) [inline docs/draft-alakuijala-brotli-07.nroff]: The last two lines should be indented to indicate that these three lines are one instruction:

if distance code is not zero, and distance is not a
   static dictionary reference, push distance to the ring
   buffer of last distances

(and similarly the "move backwards distance ..." and "look up the static dictionary word ..." instructions)

**Comment by szabadka** [comment]
Could you create a new PR with the remaining two commits? This one does not seem to be mergeable any more.

**Comment by dsnet** [comment]
Abandoning, since this is getting too messy.

**Pull request closed without merging**